Design rules/Diversen.md

```diff
@@ -0,0 +1,190 @@
+
+Bij het gebruik van 'allOf' staat de component die hergebruikt wordt altijd eerst, en staan de toegevoegde properties als tweede.
+
+Voorbeeld van het correct gebruik van 'allOf':
+
+```NaamPersoon:
+       allOf:
+         - $ref: "#/components/schemas/Naam"
+         - description: "Gegevens over de naam van de persoon"
+           properties:
+             aanhef:
+               type: "string"  
+```
+
+Voorbeeld van _**foutief**_ gebruik van 'allOf':
+```
+ 	NaamPersoon:
+ 	      allOf:
+ 	        - description: "Gegevens over de naam van de persoon"
+ 	          properties:
+ 	            aanhef:
+ 	              type: "string"
+ 	        - $ref: "#/components/schemas/Naam"
+```
+
+_**Ratio:**_ Afwijken van deze regel leidt tot problemen bij het genereren van code uit de API specificaties.
+
+Datum opname : 29-04-2021
+Datum wijziging : 29-04-2021
+
+## DR4.6 Bij het gebruik van 'allOf' is er slechts 1 component waarnaar gerefereerd wordt
+
+Bij gebruik van allOf is er altijd exact één component waarnaar gerefereerd wordt en één gedefinieerd object met ten minste één property.
+
+Voorbeeld van het foutief gebruik van allOf:
+
+```
+     NaamPersoon:
+       allOf:
+         - $ref: "#/components/schemas/Naam"
+         - $ref: "#/components/schemas/Aanschrijfwijze"
+         - description: "Gegevens over de naam van de persoon"
+           properties:
+             aanhef:
+               type: "string"
+```
+
+Er wordt hier uit twee componenten overgeërfd wat niet correct is.
+
+Voorbeeld van het foutief gebruik van allOf:
+```
+     NaamPersoon:
+       allOf:
+         - $ref: "#/components/schemas/Naam"
+         - description: "Gegevens over de naam van de persoon"
+```
+
+NaamPersoon heeft geen eigen properties wat niet correct is.
+
+Voorbeeld van correct gebruik van allOf:
+
+```
+     Naam:
+      type: "object"
+      properties:
+        geslachtsnaam:
+          type: "string"
+          description: |
+                        De achternaam van een persoon.
+          example: "Vries"
+        voorletters:
+          type: "string"
+          description: |
+                        De voorletters van de persoon, afgeleid van de voornamen.
+          example: "P.J."
+
+      NaamPersoon:
+       allOf:
+         - $ref: "#/components/schemas/Naam"
+         - description: "Gegevens over de naam van de persoon"
+           properties:
+             aanhef:
+               type: "string"
+             aanschrijfwijze:
+               type: "string"
+```
+
+_**Ratio:**_
+
+Afwijken van deze regel leidt tot problemen bij het genereren van code uit de API-specificaties.
+
+Datum opname: 15-04-2021
+Datum wijziging: 15-04-2021
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+Wanneer er meerdere componenten zijn met meerdere dezelfde properties, moet er hergebruik worden gemaakt via een 'allOf' constructie. Dit sluit aan op object oriëntatie in de verschillende programmeeromgevingen.
```

Comment:
Waar dienen de bovenstaande witregels voor?
Als deze en onderstaande regels niet bij DR4.6 horen dan wellicht een aparte Design Rule opvoeren of een paragraaf zonder Design Rule opvoeren.